lib/Serialization/ModuleFile.cpp

```diff
@@ -270,6 +300,13 @@ ValidationInfo serialization::validateSerializedAST(
                                     extendedInfo);
       if (result.status == Status::Malformed)
         return result;
+    } else if (dependencies &&
+               topLevelEntry.ID == INPUT_BLOCK_ID) {
+      cursor.EnterSubBlock(INPUT_BLOCK_ID);
+      if (validateInputBlock(cursor, scratch, *dependencies)) {
```

Comment:
We should probably only do this if `result.status` is still clear.

Reply:
Ah, good catch.